### PR TITLE
test: cover column arithmetic edge cases

### PR DESCRIPTION
**Pull request by newmattock**
/claim #560

Addresses #560

## Summary
- add focused no-default tests for column arithmetic edge cases
- cover unsigned/signed rejection paths, integer result typing/upcasts, checked overflow/division-by-zero propagation, and nonnumeric operand errors
- keep the slice scoped to `crates/proof-of-sql/src/base/database/column_arithmetic_operation.rs`, which I did not find in current open PR file lists or recent issue comments

## Verification
- `PATH="$HOME/.cargo/bin:$PATH" cargo test -p proof-of-sql --no-default-features --features="test" column_arithmetic_operation -- --nocapture`
- `PATH="$HOME/.cargo/bin:$PATH" cargo check -p proof-of-sql --no-default-features --features="test"`
- `PATH="$HOME/.cargo/bin:$PATH" cargo fmt --check`
- `git diff --check`

Generated with Codex and reviewed before submission.

### CODE REVIEW

File: crates/proof-of-sql/src/base/database/column_arithmetic_operation.rs [hunk 1]
```diff
@@ -323,3 +323,87 @@ impl ArithmeticOp for DivOp {
         try_divide_decimal_columns(lhs, rhs, left_column_type, right_column_type)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::base::scalar::test_scalar::TestScalar;
+    use alloc::{string::String, vec};
```
[the remark this answers is not in the document]
Comment (newmattock) [reply crates/proof-of-sql/src/base/database/column_arithmetic_operation.rs]: Fixed in `e8dd94d` by removing the unused `vec` import. I reran the focused column arithmetic tests plus `cargo fmt --check` and `git diff --check` before pushing.

**Comment by JayWhite2357** [comment]
Closing as part of a bulk cleanup of stale PRs opened before 2026-06-17.
